### PR TITLE
feat: Add ability to send HTML emails on build failure

### PR DESCRIPTION
**Pull request by ernestojeda**
New feature to send emails for build failures.

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Issue Number:
#312 

## Sandbox Testing
Test Links :

### CODE REVIEW

File: vars/edgeXInfraPublish.groovy [hunk 1]
```diff
@@ -65,8 +65,6 @@ def call(body) {
                 println "[edgeXInfraPublish] An error occurd while publishing logs: ${e.message}"
             }
         }
-
-        cleanWs()
```
Comment (ernestojeda) [inline vars/edgeXInfraPublish.groovy:69]: This is not needed and causes issues with files needed during the post build stages

File: src/test/groovy/edgeXBuildGoAppSpec.groovy [hunk 2]
```diff
@@ -169,7 +171,8 @@ public class EdgeXBuildGoAppSpec extends JenkinsPipelineSpecification {
                     DOCKER_BUILD_ARGS: 'MyArg1=Value1,MyArg2="Value2"',
                     ARTIFACT_ROOT: "archives/bin",
                     ARTIFACT_TYPES: 'docker',
-                    SHOULD_BUILD: true
+                    SHOULD_BUILD: true,
+                    BUILD_FAILURE_NOTIFY_LIST: 'mock@lists.edgexfoundry.org'
```
Comment (jamesrgregg) [inline src/test/groovy/edgeXBuildGoAppSpec.groovy:175]: Just wondering if this causes noise with the smtp server forcing a bounced email.

**Review by jamesrgregg — APPROVED** [review]
Excellent work @ernestojeda 
That image used in the message is huge.  Glad your CSS formats it to a good small size. 
Hope it doesn't change locations or get any bigger.

**Review by soda480 — APPROVED** [review]
LGTM

**Review by bill-mahoney — APPROVED** [review]
LGTM

**Review by cjoyv — APPROVED** [review]
LGTM

File: src/test/groovy/edgeXBuildGoAppSpec.groovy (hunk 2 above)
Comment (ernestojeda) [reply src/test/groovy/edgeXBuildGoAppSpec.groovy:175]: No, it's just mocked. No emails are actually sent when the tests are run.

**Pull request merged**